CONTRIBUTING.md

```diff
@@ -0,0 +1,257 @@
+the actual issue. If you see legitimate issues, like typos, address them in a separate commit (it's
+fine to group multiple typo fixes in a single commit).
+
+Isolate trivial refactorings into separate commits. For example, a method rename that affects dozens
+of call sites can be reviewed in a few seconds, but if it's part of a larger diff it gets mixed up
+with more complex changes (that might affect the same lines), and reviewers have to check every
+line.
+
+Commit message subjects start with a capital letter, use the imperative form and do **not** end
+with a period:
+
+* correct: "Add test for CQL request handler"
+* incorrect: "~~Added test for CQL request handler~~"
+* incorrect: "~~New test for CQL request handler~~"
+
+Avoid catch-all messages like "Minor cleanup", "Various fixes", etc. They don't provide any useful
+information to reviewers, and might be a sign that your commit contains unrelated changes.
+ 
+We don't enforce a particular subject line length limit, but try to keep it short.
+
+You can add more details after the subject line, separated by a blank line. The following pattern
+(inspired by [Netty](http://netty.io/wiki/writing-a-commit-message.html)) is not mandatory, but
+welcome for complex changes:
+
+```
+One line description of your change
+ 
+Motivation:
+
+Explain here the context, and why you're making that change.
+What is the problem you're trying to solve.
+ 
+Modifications:
+
+Describe the modifications you've done.
+ 
+Result:
+
+After your change, what will change.
+```
+
+## Pull requests
+
+Like commits, pull requests should be focused on a single, clearly stated goal.
+
+Don't base a pull request onto another one, it's too complicated to follow two branches that evolve
+at the same time. If a ticket depends on another, wait for the first one to be merged. 
+
+If you have to address feedback, avoid rewriting the history (e.g. squashing or amending commits):
+this makes the reviewers' job harder, because they have to re-read the full diff and figure out
+where your new changes are. Instead, push a new commit on top of the existing history; it will be
+squashed later when the PR gets merged. If the history is complex, it's a good idea to indicate in
+the message where the changes should be squashed:
+
+```
+* 20c88f4 - Address feedback (to squash with "Add metadata parsing logic") (36 minutes ago)
+* 7044739 - Fix various typos in Javadocs (2 days ago)
+* 574dd08 - Add metadata parsing logic (2 days ago)
+```
+
+(Note that the message refers to the other commit's subject line, not the SHA-1. This way it's still
+relevant if there are intermediary rebases.)
+
+If you need new stuff from the base branch, it's fine to rebase and force-push, as long as you don't
+rewrite the history. Just give a heads up to the reviewers beforehand. Don't push a merge commit to
+a pull request.
+
+Be sure to test your pull request in:
+
+1. Java mode
+2. Native mode
```

Comment:
```suggestion
Be sure to test your pull request by running all the unit and integration tests, including tests in native mode. Again, this can be done by running `mvn clean verify -Pnative`.
```